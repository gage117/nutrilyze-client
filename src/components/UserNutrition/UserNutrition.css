.user-nutrition {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}

.un__user-container {
  background-color: #0c7b93;
  border-radius: .75rem;
  margin: 0 auto 1rem;
  width: 29rem;
}

.un__header {
  padding: .5rem 0 .5rem 1rem;
  background-color: #27496d;
  border-radius: .75rem .75rem 0 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.un__header-text {
  font-size: 3rem;
  font-weight: bold;
  margin: 0;
  color: #40ceb8;
}

.un__clear-button {
  text-decoration: none;
  height: 3rem;
  font-size: 2.2rem;
  padding: .3rem .6rem;
  margin: 0 1rem;
  border: 1px solid #142850;
  border-radius: .3rem;
  color: whitesmoke;
  background-color: #142850;
}

.un__clear-button:hover {
  cursor: pointer;
  background-color: #40ceb8;
}

.un__h6 {
  text-align: center;
  font-size: 2.2rem;
  padding: 1rem 0;
  margin: 0 auto;
}

.un__p {
  font-size: 1.8rem;
  padding: .5rem 0;
  margin: 0;
}

.un__split-container {
  display: flex;
  justify-content: space-around;
}