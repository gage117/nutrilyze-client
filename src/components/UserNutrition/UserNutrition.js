import React from 'react'
import './UserNutrition.css'

function UserNutrition(props) {
  function handleDeleteClick() {
    props.handlers.deleteUserNutrition(props.state.users[0])
  }

  const users = props.state.users.map((user, index) => {
    return (
      <div className='un__user-container' key={index}>
        <div className='un__header'>
          <span className='un__header-text'>{user.name}</span>
          <button className='un__clear-button' onClick={handleDeleteClick}>Clear</button>
        </div>
        <h6 className='un__h6'>Calories: {user.calories}</h6>
        <div className='un__split-container'>
          <section className='un__half'>
            <p className='un__p'>Protein: {user.protein}g</p>
            <p className='un__p'>Carbs: {user.carbs}g</p>
            <p className='un__p'>Sugar: {user.sugar}g</p>
          </section>
          <section className='un__half'>
            <p className='un__p'>Fiber: {user.fiber}g</p>
            <p className='un__p'>Fat: {user.fat}g</p>
            <p className='un__p'>Sodium: {user.sodium}mg</p>
          </section>
        </div>
      </div>
    )
  })
  return (
    <div className='user-nutrition'>
      {users}
    </div>
  )
}

export default UserNutrition
