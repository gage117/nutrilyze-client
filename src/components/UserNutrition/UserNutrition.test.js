import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'
import UserNutrition from './UserNutrition'

describe('UserNutrition', () => {
  const state = {
    users: [
      {
        id: 1,
        name: 'Gage',
        calories: 666,
        protein: 20,
        carbs: 40,
        sugar: 20,
        fiber: 10,
        fat: 30,
        sodium: 20
      }
    ]
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <UserNutrition state={state}/>
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders the UI as exptected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <UserNutrition state={state}/>
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})