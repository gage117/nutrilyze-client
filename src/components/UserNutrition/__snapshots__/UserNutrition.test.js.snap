// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNutrition renders the UI as exptected 1`] = `
<div
  className="user-nutrition"
>
  <div
    className="un__user-container"
  >
    <h3
      className="un__h3"
    >
      Gage
    </h3>
    <h6
      className="un__h6"
    >
      Calories: 
      666
    </h6>
    <div
      className="un__split-container"
    >
      <section
        className="un__half"
      >
        <p
          className="un__p"
        >
          Protein: 
          20
          g
        </p>
        <p
          className="un__p"
        >
          Carbs: 
          40
          g
        </p>
        <p
          className="un__p"
        >
          Sugar: 
          20
          g
        </p>
      </section>
      <section
        className="un__half"
      >
        <p
          className="un__p"
        >
          Fiber: 
          10
          g
        </p>
        <p
          className="un__p"
        >
          Fat: 
          30
          g
        </p>
        <p
          className="un__p"
        >
          Sodium: 
          20
          mg
        </p>
      </section>
    </div>
  </div>
</div>
`;
